### PR TITLE
Add `update_presence` to bot

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/bot.py [hunk 1]
```diff
@@ -59,6 +59,7 @@
 from hikari.utilities import date
 from hikari.utilities import event_stream
 from hikari.utilities import ux
```
Comment (nekokatt) [inline hikari/impl/bot.py:61]: please run isort if you havent already

I would assume `ux` comes after `un`, but might be wrong

File: hikari/impl/bot.py [hunk 2]
```diff
@@ -823,6 +824,56 @@ async def wait_for(
...
+
+            This method is simply a facade to make performing this in bulk
+            simpler.
+        """
+        coros = [
+            s.update_presence(status=status, activity=activity, idle_since=idle_since, afk=afk)
+            for s in self._shards.values()
+        ]
+
+        await asyncio.gather(*coros)
```
Comment (nekokatt) [inline hikari/impl/bot.py:875]: Use aio.all_of here or you will get incomplete task spam possibly if something fails due to a cancelled error.

File: hikari/impl/bot.py [hunk 3]
```diff
@@ -823,6 +824,56 @@ async def wait_for(
     ) -> event_dispatcher.EventT_co:
         return await self._events.wait_for(event_type, timeout=timeout, predicate=predicate)
 
+    async def update_presence(
```
Comment (FasterSpeeding) [inline hikari/impl/bot.py:827]: Would it not make sense for this method to maybe also be declared on shard aware for the traits?

Comment (davfsa) [reply hikari/impl/bot.py:827]: All the traits only include properties, not public methods, which is why I didnt think about adding it. @nekokatt?

Comment (nekokatt) [reply hikari/impl/bot.py:827]: It should be added.

Comment (davfsa) [reply hikari/impl/bot.py:827]: Added :+1:

**Pull request merged**